src/main/holochainManager.ts

```diff
@@ -250,21 +263,85 @@ export class HolochainManager {
 
   // TODO Add option to install happ without UI
   async installWebHapp(webHappPath: string, appId: string, networkSeed?: string) {
-    const uiTargetDir = this.fs.happUiDir(appId, this.holochainDataRoot);
-    console.log('uiTargetDir: ', uiTargetDir);
-    console.log('Installing app...');
-    const tempHappPath = await rustUtils.saveWebhapp(webHappPath, uiTargetDir);
-    console.log('Stored UI and got temp happ path: ', tempHappPath);
+    // Decode webhapp into .happ bytes and ui.zip bytes
+    // TODO Option to install headless app
+    const happAndUiBytes = await rustUtils.readAndDecodeWebhapp(webHappPath);
+
+    // write [sha256].happ to happs directory
+    const happHasher = crypto.createHash('sha256');
+    const happSha256 = happHasher.update(Buffer.from(happAndUiBytes.happBytes)).digest('hex');
+    const happFilePath = path.join(this.fs.happsDir(this.holochainDataRoot), `${happSha256}.happ`);
+    writeFile(happFilePath, Buffer.from(happAndUiBytes.happBytes));
+
+    // compute UI hash
+    const uiZipHasher = crypto.createHash('sha256');
+    const uiZipSha256 = uiZipHasher.update(Buffer.from(happAndUiBytes.uiBytes)).digest('hex');
+
+    // If same UI is not already stored, write ui bytes to temp file to compute hashes
+    // (randomly generated filename to prevent other processes from being able to modify
```

Comment:
Correct, it just makes it harder. Do you think that's not good enough? The hashing happens immediately after unpacking so I imagine it would be pretty hard to tamper here before the hashes are computed. But maybe there would be a way to already compute the hashes in memory based off the bytes onlly, I'm not sure.

Reply:
I think this is fine, I'd just adjust the comment to make the claim less strong